catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/CoordinateProcessor.java

```diff
@@ -0,0 +1,181 @@
+      enhanced.addFirst(literal);
+    }
+    return enhanced;
+  }
+
+  private LiteralSuggestion getLatLonSuggestions(String query) {
+    Matcher matcher = PATTERN_LAT_LON.matcher(query);
+    List<Double> numbers = new ArrayList<>();
+    while (matcher.find()) {
+      numbers.add(Double.parseDouble(matcher.group()));
+    }
+    if (numbers.isEmpty() || numbers.size() == 1) {
+      return null;
+    }
+    List<LatLon> latLonLists = getLatLonLists(numbers);
+    String name = "Lat/Lon: ";
+    for (int index = 0; index < latLonLists.size(); index++) {
+      if (index == latLonLists.size() - 1) {
+        name += latLonLists.get(index).toString();
+      } else {
+        name += latLonLists.get(index).toString() + ", ";
+      }
+    }
+    return new LiteralSuggestion("LITERAL", name, latLonLists);
+  }
+
+  private List<LatLon> getLatLonLists(List<Double> numbers) {
+    List<LatLon> latLons = new ArrayList<>();
+    for (int index = 0; index < numbers.size(); index++) {
+      double latitude = numbers.get(index);
+      if (isValidLatitude(latitude)) {
+        if (index + 1 < numbers.size()) {
+          double longitude = numbers.get(index + 1);
+          if (isValidLongitude(longitude)) {
+            latLons.add(LatLon.from(latitude, longitude));
+            ++index;
+          }
+        }
+      }
+    }
+    return latLons;
+  }
+
+  private boolean isValidLatitude(double latitude) {
+    return latitude >= -90 && latitude <= 90;
+  }
+
+  private boolean isValidLongitude(double longitude) {
+    return longitude >= -180 && longitude <= 180;
+  }
+
+  @VisibleForTesting
+  static class LiteralSuggestion implements Suggestion {
+    private final String id;
+    private final String name;
+    private final List<LatLon> geo;
+
+    LiteralSuggestion(String id, String name, List<LatLon> geo) {
+      this.id = id;
+      this.name = name;
+      this.geo = geo;
+    }
+
+    @Override
+    public String getId() {
+      return id;
+    }
+
+    @Override
+    public String getName() {
+      return name;
+    }
+
+    public List<LatLon> getGeo() {
+      return geo;
+    }
+
+    public boolean hasGeo() {
+      return geo != null && !geo.isEmpty();
+    }
+  }
+
+  @VisibleForTesting
+  static class LatLon {
+    private final Double lat;
+    private final Double lon;
+
+    static LatLon from(Double lat, Double lon) {
+      return new LatLon(lat, lon);
+    }
+
+    private LatLon(Double lat, Double lon) {
+      notNull(lat);
+      notNull(lon);
+      this.lat = lat;
+      this.lon = lon;
+    }
+
+    public Double getLat() {
+      return lat;
+    }
+
+    public Double getLon() {
+      return lon;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      LatLon latLon = (LatLon) o;
+
+      if (!lat.equals(latLon.lat)) {
```

Comment:
✏️ This if can be squashed into a single statement like `return lat.equals(latLon.lat) && lon.equals(latLon.lon);`.